src/VCF_Tasks.cpp

```diff
@@ -51,4 +51,37 @@ bool run_read_adc2_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo&
 
     return true;
 }
+
+bool init_read_gpio_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo)
+{
+    // Setting digital/analog buttons D10-D6, A8 as inputs
+    pinMode(BTN_DIM_READ, INPUT);
+    pinMode(BTN_PRESET_READ, INPUT);
+    pinMode(BTN_MC_CYCLE_READ, INPUT);
+    pinMode(BTN_MODE_READ, INPUT);
+    pinMode(BTN_START_READ, INPUT);
+    // pinMode(BTN_DATA_READ, INPUT); this is an analog pin, so technically no need for pinMode as analogRead will setup the pin
+    
+    return true;
+}
+bool run_read_gpio_task(const unsigned long& sysMicros, const HT_TASK::TaskInfo& taskInfo)
+{
+    // Doing digital read on all digital inputs
+    int dimButton = digitalRead(BTN_DIM_READ);
+    int presetButton = digitalRead(BTN_PRESET_READ);
+    int mcCycleButton = digitalRead(BTN_MC_CYCLE_READ);
+    int modeButton = digitalRead(BTN_MODE_READ);
+    int startButton = digitalRead(BTN_START_READ);
+    int dataButton = analogRead(BTN_DATA_READ); // this is on an analog pin, so technically need to use analogRead
```

Comment:
Teensy has multiple types of pins. Some pins can only be used as digital inputs, some can be also be used as analog inputs, etc. Even though this is an "analog" pin, it's really just an "analog-capable" pin. We're still using it as a digital pin, so please treat it the same as the other pins.